integration_testing/features/super-admin/super-admin-hard-delete-user.feature

```diff
@@ -0,0 +1,17 @@
+Feature: Super Admin delete user
+    Super Admin need to be able to delete a user from the command line
+
+  Background:
+    Given that there exists a user with username <username> I want to delete
+      And I have access to terminal for Kolibri
+
+  Scenario: Delete user with username <username>
+    When I type `kolibri manage deleteuser <username>` into the terminal
+      And I hit *Enter*
+    Then I see "Are you sure you wish to permanently delete this user? This will DELETE ALL DATA FOR THIS USER." message
+    When I type "yes"
+      And I hit *Enter*
+    Then I see "ARE YOU SURE? If you do this, there is no way to recover the user data on this device." message
+    When I type "yes"
+      And I hit *Enter*
+    Then all user data associated with username <username> should be deleted
```

Comment:
Still trying to figure out what's the best way to comment on these... 😕 
Suggestion:

```
Feature: Super Admin delete user
    Super Admin needs to be able to delete a user from the command line

  Background:
    Given that there exists a user with username <username> I want to delete
      And I have access to Terminal on the device

  Scenario: Delete user and all their data
    When I type `kolibri manage deleteuser <username>` in the Terminal
      And I press *Enter* key
    Then I see "Are you sure you wish to permanently delete this user? This will DELETE ALL DATA FOR THIS USER." message
    When I type "yes"
      And I press *Enter* key
    Then I see "ARE YOU SURE? If you do this, there is no way to recover the user data on this device." message
    When I type "yes"
      And I press *Enter* key
    ~~Then all user data associated with username <username> should be deleted~~

Examples:
| username |
| coach    |
| learner  |
| admin    |
```

- Line 2: added 's'
- Line 6: capital 'T', 'access _on device_'
- Line 8: rephrased (no need for the placeholder in the scenario title)
- Lines 10/13/16: 'press' instead of 'hit', added 'key'
- Line 12/15: Is it 'yes' or 'y' would suffice too?
- Line 17: try to avoid "Then's" with 'should', as it is not a testable condition. Is there a confirmation message or just the prompt? If latter, than add a "When" with how to test if the deletion has been successful, for example:
```
When I go to *Facility > Users*
  And I search for <username>
Then I see no results
```
- Lines 19-23: added values for placeholders

Reply:
in gherkin stories, is it best practice to copy the user-facing strings verbatim and make them part of what's being tested? Or would it be preferable to more abstractly describe the interaction.

E.g. 

> Then I see "ARE YOU SURE? If you do this, there is no way to recover the user data on this device." message

versus

> Then I see a second confirmation message, reiterating that I cannot recover the user data.

The latter would be a bit less "brittle" and focus the test on the essential user flow rather than little details.

Reply:
Yeah, I think copy should be evaluated in another context